doc/developers/contributing.rst

```diff
@@ -268,100 +253,118 @@ If any of the above seems like magic to you, then look up the `Git documentation
-  Examples should demonstrate why the new functionality is useful in
-  practice and, if possible, compare it to other methods available in
-  scikit-learn.
-
-* Documentation and high-coverage tests are necessary for enhancements to be
-  accepted. Bug-fixes or new features should be provided with
-  `non-regression tests
-  <https://en.wikipedia.org/wiki/Non-regression_testing>`_. These tests
-  verify the correct behavior of the fix or feature. In this manner, further
-  modifications on the code base are granted to be consistent with the
-  desired behavior. For the case of bug fixes, at the time of the PR, the
-  non-regression tests should fail for the code base in the master branch
-  and pass for the PR code.
-
-* At least one paragraph of narrative documentation with links to
-  references in the literature (with PDF links when possible) and
-  the example. For more details on writing and building the
-  documentation, see the :ref:`contribute_documentation` section.
-
-* The documentation should also include expected time and space complexity
-  of the algorithm and scalability, e.g. "this algorithm can scale to a
-  large number of samples > 100000, but does not scale in dimensionality:
-  n_features is expected to be lower than 100".
+Before a PR can be merged, it needs to be approved by two core developers.
+Please prefix the title of your pull request with ``[MRG]`` if the
+contribution is complete and should be subjected to a detailed review. An
+incomplete contribution -- where you expect to do more work before receiving
+a full review -- should be prefixed ``[WIP]`` (to indicate a work in
+progress) and changed to ``[MRG]`` when it matures. WIPs may be useful to:
+indicate you are working on something to avoid duplicated work, request
+broad review of functionality or API, or seek collaborators. WIPs often
+benefit from the inclusion of a `task list
+<https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments>`_ in
+the PR description.
+
+In order to ease the reviewing process, we recommend that your contribution
+complies with the following rules before marking a PR as ``[MRG]``. The
+**bolded** ones are especially important:
+
+
+1. **Make sure your code passes the tests**. The whole test suite can be run
+   with `pytest`, but it is usually not recommended since it takes a long
+   time. It is often enough to only run the test related to your changes:
+   for example, if you changed something in
+   `sklearn/linear_model/logistic.py`, running the following commands will
+   usually be enough:
+
+   - `pytest sklearn/linear_model/logistic.py` to make sure the doctest
+     examples are correct
+   - `pytest sklearn/linear_model/tests/test_logistic.py` to run the tests
+     specific to the file
+   - `pytest sklearn/linear_model` to test the whole `linear_model` module
+   - `pytest sklearn/doc/linear_model.rst` to make sure the user guide
+     examples are correct.
```

Comment:
adding the test_common and test_docstrings may also be useful?

Reply:
Adding test_common, but test_docstrings very rarely fails so probably not worth adding it